internal/pkg/agent/storage/store/state_store_test.go

```diff
@@ -33,14 +40,542 @@ func TestStateStore(t *testing.T) {
 	t.Run("no ack token", func(t *testing.T) {
 		runTestStateStore(t, "")
 	})
+
+	t.Run("migrate", func(t *testing.T) {
+		if runtime.GOOS == "darwin" {
+			// the original test never actually run, so with this at least
+			// there is coverage for linux and windows.
+			t.Skipf("needs https://github.com/elastic/elastic-agent/issues/3866" +
+				"to be merged so this test can work on darwin")
+		}
+
+		t.Run("action store file does not exists", func(t *testing.T) {
+			ctx := context.Background()
```

Comment:
We could use a context that expires after a while, just in case the test gets stuck...

Reply:
I could do that, but those tests do not wait on anything, there isn't any expectation they might get stuck.